inc/notificationtargetentity.class.php

```diff
@@ -276,5 +360,46 @@ public static function uninstall()
 
             $template->delete($template_data);
         }
+
+        $notifications_iterator_quantity = $DB->request(
+            [
+                'SELECT' => 'id',
+                'FROM' => $notification->getTable(),
+                'WHERE' => [
+                    'itemtype' => 'PluginCreditEntity',
+                    'event' => 'lowcredits',
+                ],
+            ]
+        );
+        foreach ($notifications_iterator_quantity as $notification_data_quantity) {
+            $notification->delete($notification_data_quantity);
+        }
+
+        $templates_iterator_quantity = $DB->request(
+            [
+                'SELECT' => 'id',
+                'FROM' => $template->getTable(),
+                'WHERE' => [
+                    'itemtype' => 'PluginCreditEntity',
+                    'name' => 'lowcredits',
+                ],
+            ]
+        );
+        foreach ($templates_iterator_quantity as $template_data_quantity) {
+            $translations_iterator_quantity = $DB->request(
+                [
+                    'SELECT' => 'id',
+                    'FROM' => $translation->getTable(),
```

Comment:
I don't understand why this error is returned

Reply:
Because it's previously declared in a `for` loop, and it's possible code never enter in the loop, ending with an undefined variable.
Best way is to declare it where you need.